---
title: "DADA2_Ejemplo_Clase"
author: "Roberto Álvarez"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
    html_document: 
      fig_height: 8
      fig_width: 13
      number_section: yes
      theme: cerulean
      toc: yes
      toc_float:
        collapsed: no
        smooth_scroll: yes
---
```{r message=FALSE, warning=FALSE}
library(dada2)
```

```{r}
path <- "~/Dropbox/DADA2_Ejemplo_Clase/MiSecuenciasAqui/" # CHANGE ME to the directory containing the fastq files after unzipping.
list.files(path)
```

```{r}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)

```

## Calidad phred

```{r}
plotQualityProfile(fnFs[1:2])
```

```{r}
plotQualityProfile(fnRs[1:2])
```


## Filtrado y trimming

```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```


```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
head(out)
```


## Filtrado y tasas de error

```{r}
errF <- learnErrors(filtFs, multithread=TRUE)
#save(errF,file="~/Dropbox/DADA2_Ejemplo_Clase/errF.RData")
```


```{r}
errR <- learnErrors(filtRs, multithread=TRUE)
#save(errR,file="~/Dropbox/DADA2_Ejemplo_Clase/errR.RData")
```


```{r}
plotErrors(errF, nominalQ=TRUE)
```


## Inferencia de la muestra



```{r}
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
#save(dadaFs, file="~/Dropbox/DADA2_Ejemplo_Clase/dadaFs.RData")
```

```{r}
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
#save(dadaRs, file="~/Dropbox/DADA2_Ejemplo_Clase/dadaRs.RData")
```


```{r}
dadaFs[[1]]
```


## Merge paired reads



```{r}
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
```


## Construir la tabal de secuencias

```{r}
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```


## Remover quimeras


```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
```

```{r}
sum(seqtab.nochim)/sum(seqtab)
```


## Verificar el número de lecturas después del pipeline


```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
```

  
## Asignar taxonomía

Hay que descragar las bases de datos más actualizada acá en la siguiente liga (Taxonomic reference data)[https://benjjneb.github.io/dada2/training.html] . Los archivos que descargan son dos: silva_nr_v132_train_set.fa.gz y 

Estos dos procesos pueden tardar mucho, cuidado


```{r}
taxa <- assignTaxonomy(seqtab.nochim,"~/Dropbox/DADA2_Ejemplo_Clase/tax/silva_nr_v132_train_set.fa.gz", multithread=TRUE)
#save(taxa,file="~/Dropbox/DADA2_Ejemplo_Clase/taxa1.RData")
```


```{r}
taxa <- addSpecies(taxa,"~/Dropbox/DADA2_Ejemplo_Clase/tax/silva_species_assignment_v132.fa.gz")
#save(taxa,file="~/Dropbox/DADA2_Ejemplo_Clase/taxa.RData")
```

```{r}
taxa.print <- taxa # Removing sequence rownames for display only
rownames(taxa.print) <- NULL
head(taxa.print)
```

### Alternativas (OPCIONAL)

```{r}
# Si no tienen instalado esta librería pueden hacerlo
# BiocManager::install("DECIPHER")
#library(DECIPHER); packageVersion("DECIPHER")
```

NO EJECUTAR SI NO TIENES BIEN INSTALADO DECIPHER

```{r}
#dna <- DNAStringSet(getSequences(seqtab.nochim)) # Create a DNAStringSet from the ASVs
#load("~/tax/IDTaxa/SILVA_SSU_r132_March2018.RData") # CHANGE TO THE PATH OF YOUR TRAINING SET
#ids <- IdTaxa(dna, trainingSet, strand="top", processors=NULL, verbose=FALSE) # use all processors
#ranks <- c("domain", "phylum", "class", "order", "family", "genus", "species") # ranks of interest
# Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy
#taxid <- t(sapply(ids, function(x) {
#        m <- match(ranks, x$rank)
#        taxa <- x$taxon[m]
#        taxa[startsWith(taxa, "unclassified_")] <- NA
 #       taxa
#}))
#colnames(taxid) <- ranks; rownames(taxid) <- getSequences(seqtab.nochim)
```


## Evaluar precisión


```{r}
unqs.mock <- seqtab.nochim["Mock",]
unqs.mock <- sort(unqs.mock[unqs.mock>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock), "sample sequences present in the Mock community.\n")
```



```{r}
mock.ref <- getSequences(file.path(path, "HMP_MOCK.v35.fasta"))
match.ref <- sum(sapply(names(unqs.mock), function(x) any(grepl(x, mock.ref))))
cat("Of those,", sum(match.ref), "were exact matches to the expected reference sequences.\n")

```


## Convertir a phyloseq

```{r message=FALSE, warning=FALSE}
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
```


```{r}
theme_set(theme_bw())
```


```{r}
samples.out <- rownames(seqtab.nochim)
subject <- sapply(strsplit(samples.out, "D"), `[`, 1)
gender <- substr(subject,1,1)
subject <- substr(subject,2,999)
day <- as.integer(sapply(strsplit(samples.out, "D"), `[`, 2))
samdf <- data.frame(Subject=subject, Gender=gender, Day=day)
samdf$When <- "Early"
samdf$When[samdf$Day>100] <- "Late"
rownames(samdf) <- samples.out
```



```{r}
ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(samdf), 
               tax_table(taxa))
ps <- prune_samples(sample_names(ps) != "Mock", ps) # Remove mock sample
```



```{r}
dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
ps
```

# Al fin deiversidad

```{r}
plot_richness(ps, x="Day", color="When")

```


# Algunas medidas de reducción de la dimensionalidad

```{r}
# Transform data to proportions as appropriate for Bray-Curtis distances
ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu))
ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray")
```

```{r}
plot_ordination(ps.prop, ord.nmds.bray, color="When", title="Bray NMDS")
```


# Gráficos de barras apiladas de abundancias

```{r}
top20 <- names(sort(taxa_sums(ps), decreasing=TRUE))[1:20]
ps.top20 <- transform_sample_counts(ps, function(OTU) OTU/sum(OTU))
ps.top20 <- prune_taxa(top20, ps.top20)
plot_bar(ps.top20, x="Day", fill="Family") + facet_wrap(~When, scales="free_x")
```

